pandas/core/arrays/boolean.py

```diff
@@ -200,7 +205,7 @@ def coerce_to_array(
         if inferred_dtype not in ("boolean", "empty") + integer_like:
             raise TypeError("Need to pass bool-like values")
 
-        mask_values = isna(values_object)
+        mask_values = cast("npt.NDArray[np.bool_]", isna(values_object))
```

Comment:
Add comment why the cast is necessary.

The other possible solution here is to use `mask_values1` here through lines 219, and then add line 220 that says `mask_values = mask_values1`, but that's more of a code change.